docs/source/tasks/asr.mdx

```diff
@@ -0,0 +1,222 @@
+
+```py
+>>> timit = timit.map(prepare_dataset, remove_columns=timit.column_names["train"], num_proc=4)
+```
+
+🤗 Transformers doesn't have a data collator for automatic speech recognition, so you will need to create one. You can adapt the [`DataCollatorWithPadding`] to create a batch of examples for automatic speech recognition. It will also dynamically pad your text and labels to the length of the longest element in its batch, so they are a uniform length. While it is possible to pad your text in the `tokenizer` function by setting `padding=True`, dynamic padding is more efficient.
+
+Unlike other data collators, `DataCollatorWithPadding` needs to apply a different padding method to `input_values` and `labels`. You can apply a different padding method with a context manager:
+
+```py
+>>> import torch
+
+>>> from dataclasses import dataclass, field
+>>> from typing import Any, Dict, List, Optional, Union
+
+
+>>> @dataclass
+... class DataCollatorCTCWithPadding:
+
+...     processor: AutoProcessor
+...     padding: Union[bool, str] = True
+...     max_length: Optional[int] = None
+...     max_length_labels: Optional[int] = None
+...     pad_to_multiple_of: Optional[int] = None
+...     pad_to_multiple_of_labels: Optional[int] = None
+
+...     def __call__(self, features: List[Dict[str, Union[List[int], torch.Tensor]]]) -> Dict[str, torch.Tensor]:
+...         # split inputs and labels since they have to be of different lengths and need
+...         # different padding methods
+...         input_features = [{"input_values": feature["input_values"]} for feature in features]
+...         label_features = [{"input_ids": feature["labels"]} for feature in features]
+
+...         batch = self.processor.pad(
+...             input_features,
+...             padding=self.padding,
+...             max_length=self.max_length,
+...             pad_to_multiple_of=self.pad_to_multiple_of,
+...             return_tensors="pt",
+...         )
+...         with self.processor.as_target_processor():
+...             labels_batch = self.processor.pad(
+...                 label_features,
+...                 padding=self.padding,
+...                 max_length=self.max_length_labels,
+...                 pad_to_multiple_of=self.pad_to_multiple_of_labels,
+...                 return_tensors="pt",
+...             )
+
+...         # replace padding with -100 to ignore loss correctly
+...         labels = labels_batch["input_ids"].masked_fill(labels_batch.attention_mask.ne(1), -100)
+
+...         batch["labels"] = labels
+
+...         return batch
+```
+
+Create a batch of examples and dynamically pad them with `DataCollatorForCTCWithPadding`:
+
+```py
+>>> data_collator = DataCollatorCTCWithPadding(processor=processor, padding=True)
+```
+
+# Fine-tune with Trainer
+
+Load Wav2Vec2 with [`AutoModelForCTC`]. Save GPU memory by calculating the mean of the loss reduction instead of the sum, and specify the padding token:
```

Comment:
```suggestion
Load Wav2Vec2 with [`AutoModelForCTC`]. It's often better to average the CTC loss instead of using the default configuration of CTC loss summation:
```

No memory is saved by doing `"mean"` - setting the default to `"sum"` was a bad copy-paste from `fairseq` that we can't change anymore really due to backwards breaking